File: proj8/proj_8_testers/tests/test3.ml
# Tests ability to return values

begin
    foo1() > int begin
        return 10;
    end

    foo2() > string begin
        return "Winter is coming.\n";
    end

    foo3() > string begin
        int x;
        read(x);
        if x >= 10 then begin
            return "Yes\n";
        end else begin
            return "No\n";
        end
    end

    foo4() > bool begin
        return True;
    end

    write(foo1(), "\n");
    write(5*foo1() + foo1(), "\n");
    write(foo2());
    write(foo3());
    if foo4() then begin
        write("Yes\n");
    end
end

File: archive/proj8/proj_8_testers/c_files/test3.c
// Test ability to return values.
#include <stdio.h>

#define true 0   // This will be unnecessary for you.
#define bool int // As will this.

int foo1() {
  return 10;
}

char* foo2() {  // You will, of course, use your string type instead of a char* type.
  return "Winter is coming.\n";
}

char* foo3() {
  int x;
  scanf("%d", &x);
  if (x >= 10) {
    return "Yes\n";
  }
  else {
    return "No\n";
  }
}

bool foo4() {
  return true;
}


int main() {
  printf("%d\n", foo1());
  printf("%d\n", 5*foo1() + foo1());
  printf("%s", foo2());
  printf("%s", foo3());
  if (foo4()) {
    printf("Yes\n");
  }
}
